drivers/char/rust_example.rs

```diff
@@ -125,6 +128,59 @@ impl KernelModule for RustExample {
             cv.free_waiters();
         }
 
+        // Test threads.
+        {
+            let mut a = 1;
+            // XXX use a completion or Barrier
+            let flag = Arc::try_new(AtomicBool::new(false))?;
+            let other = flag.clone();
+
+            let t1 = Thread::new(
+                move || {
+                    other.store(true, Ordering::Release);
+                    let b = Box::try_new(42)?;
+                    for _ in 0..20 {
+                        a += 1;
+                        println!("Hello Rust Thread {}", a + b.as_ref());
+                    }
+
+                    Ok(())
+                },
+                cstr!("Rust thread"),
```

Comment:
Nit: it looks like putting the thread name as the first parameter would be preferable here because then the last (and usually longest one) comes at the end.

Reply:
Done in v3